From: Kassey Li <quic_yingangl@quicinc.com>
Subject: mm/page_owner/cma: show pfn in cma/page_owner with hex format
Date: Tue, 13 Jun 2023 17:25:33 +0800

cma: display pfn as well as pfn_to_page(pfn)

page_owner: display pfn in hex rather than decimal

Link: https://lkml.kernel.org/r/20230613092533.15449-1-quic_yingangl@quicinc.com
Signed-off-by: Kassey Li <quic_yingangl@quicinc.com>
Cc: Joonsoo Kim <iamjoonsoo.kim@lge.com>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/cma.c        |    4 ++--
 mm/page_owner.c |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/mm/cma.c~mm-page_owner-cma-show-pfn-in-cma-page_onwer-with-hex-format
+++ a/mm/cma.c
@@ -483,8 +483,8 @@ struct page *cma_alloc(struct cma *cma,
 		if (ret != -EBUSY)
 			break;
 
-		pr_debug("%s(): memory range at %p is busy, retrying\n",
-			 __func__, pfn_to_page(pfn));
+		pr_debug("%s(): memory range at pfn 0x%lx %p is busy, retrying\n",
+			 __func__, pfn, pfn_to_page(pfn));
 
 		trace_cma_alloc_busy_retry(cma->name, pfn, pfn_to_page(pfn),
 					   count, align);
--- a/mm/page_owner.c~mm-page_owner-cma-show-pfn-in-cma-page_onwer-with-hex-format
+++ a/mm/page_owner.c
@@ -418,7 +418,7 @@ print_page_owner(char __user *buf, size_
 	pageblock_mt = get_pageblock_migratetype(page);
 	page_mt  = gfp_migratetype(page_owner->gfp_mask);
 	ret += scnprintf(kbuf + ret, count - ret,
-			"PFN %lu type %s Block %lu type %s Flags %pGp\n",
+			"PFN 0x%lx type %s Block %lu type %s Flags %pGp\n",
 			pfn,
 			migratetype_names[page_mt],
 			pfn >> pageblock_order,
_
